### PR TITLE
complete assignment 4

### PR DESCRIPTION
**Pull request by ppzhuk**

### CODE REVIEW

**Comment by ppzhuk** [comment]
A little bit late, just finished the assignment.

File: app/src/main/java/ru/ppzh/lesson4/FeedActivity.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                } finally {
+                    if (cursor != null)
+                        cursor.close();
+                }
+                return articles;
+            }
+
+            private Article[] getArticles(Cursor cursor) {
+                int count = cursor.getCount();
+                cursor.moveToNext();
```
Comment (thevery) [inline app/src/main/java/ru/ppzh/lesson4/FeedActivity.java:44]: why not just `while (cursor.moveToNext())`?

File: app/src/main/java/ru/ppzh/lesson4/RssProvider.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    private static final String ITEM = "item";
+
+    public static final String TITLE = "title";
+    public static final String DATE = "pubDate";
+    public static final String DESCRIPTION = "description";
+    public static final String NEWS_URL = "link";
+
+    @Override
+    public boolean onCreate() {
+        return false;
```
Comment (thevery) [inline app/src/main/java/ru/ppzh/lesson4/RssProvider.java:28]: why false?

File: app/src/main/java/ru/ppzh/lesson4/RssProvider.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    }
+
+    @Nullable
+    @Override
+    public Cursor query(Uri uri, String[] projection, String selection, String[] selectionArgs, String sortOrder) {
+        MatrixCursor cursor = new MatrixCursor(new String[]{TITLE, DATE, DESCRIPTION, NEWS_URL});
+        String title = "", date = "", description = "", news_url = "";
```
Comment (thevery) [inline app/src/main/java/ru/ppzh/lesson4/RssProvider.java:35]: use 1 line per declaration, please

File: app/src/main/java/ru/ppzh/lesson4/RssProvider.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        Log.i(TAG, "uri: " + selection);
+        try {
+            URL url = new URL(selection);
+            XmlPullParser xpp = XmlPullParserFactory.newInstance().newPullParser();
+            xpp.setInput(url.openStream(), null);
```
Comment (thevery) [inline app/src/main/java/ru/ppzh/lesson4/RssProvider.java:40]: close?

**Comment by thevery** [comment]
10 + 1 - 1 (code issues) = 10

**Pull request closed without merging**